HPC/Gromacs.md

```diff
@@ -0,0 +1,112 @@
+* Hpc7g (-DGMX_SIMD=ARM_SVE)
+* c6gn (-DGMX_SIMD=ARM_NEON_ASIMD)
+* c5n/c6i (-DGMX_SIMD=AVX_512)
+* hpc6a (-DGMX_SIMD=AVX2_256)
+We noticed that on the same Hpc7g instance, which supports both SVE and NEON, choose ARM_SVE as SIMD option gives 10% better performance than ARM_NEON_ASIMD. 
+
+## Build Gromacs
+* build Gromacs with Armpl fft library
+```
+wget -q http://ftp.gromacs.org/pub/gromacs/gromacs-2022.4.tar.gz
+export ROOT=/shared
+CURDIR=${ROOT}/gromacs-2022.4
+export GCC_VERSION=12.2.0
+export PATH=${ROOT}/gcc-${GCC_VERSION}/bin:$PATH
+export LD_LIBRARY_PATH=${ROOT}/gcc-${GCC_VERSION}/lib64:$LD_LIBRARY_PATH
+export PATH=/shared/binutils-2.35/bin:$PATH
+export LD_LIBRARY_PATH=/shared/binutils-2.35/lib:$LD_LIBRARY_PATH
+export PATH=/shared/openmpi-4.1.4/bin:$PATH
+export LD_LIBRARY_PATH=/shared/openmpi-4.1.4/lib:$LD_LIBRARY_PATH
+MODULEPATH=/shared/arm/modulefiles module load armpl/22.1.0_gcc-10.2
+
+export LDFLAGS="-lgfortran -lamath -lm -lastring"
+cmake .. -DGMX_BUILD_OWN_FFTW=OFF \
+      -DREGRESSIONTEST_DOWNLOAD=ON \
+      -DCMAKE_C_FLAGS="-mcpu=neoverse-512tvb --param=aarch64-autovec-preference=4 -g" \
+      -DCMAKE_CXX_FLAGS="-mcpu=neoverse-512tvb --param=aarch64-autovec-preference=4 -g" \
+      -DCMAKE_C_COMPILER=$(which mpicc) \
+      -DCMAKE_CXX_COMPILER=$(which mpicxx) \
+      -DGMX_OMP=ON \
+      -DGMX_MPI=ON \
+      -DGMX_SIMD=ARM_SVE \
+      -DGMX_BUILD_MDRUN_ONLY=OFF \
+      -DGMX_DOUBLE=OFF \
+      -DCMAKE_INSTALL_PREFIX=${CURDIR} \
+      -DBUILD_SHARED_LIBS=OFF \
+      -DGMX_FFT_LIBRARY=fftw3 \
+      -DFFTWF_LIBRARY=${ARMPL_LIBRARIES}/libarmpl_lp64.so \
+      -DFFTWF_INCLUDE_DIR=${ARMPL_INCLUDES} \
+      \
+      -DGMX_BLAS_USER=${ARMPL_LIBRARIES}/libarmpl_lp64.so \
+      -DGMX_LAPACK_USER=${ARMPL_LIBRARIES}/libarmpl_lp64.so \
+      \
+      -DGMX_GPLUSPLUS_PATH=$(which g++) \
+      -DGMXAPI=OFF \
+      -DGMX_GPU=OFF
+
+make 
+make  install
+cd ..
+```
+* configure Gromacs with FFTW
+```
+cmake .. -DGMX_BUILD_OWN_FFTW=OFF \
+      -DREGRESSIONTEST_DOWNLOAD=ON \
+      -DCMAKE_C_FLAGS="-mcpu=native --param=aarch64-autovec-preference=4 -g" \
+      -DCMAKE_CXX_FLAGS="-mcpu=native --param=aarch64-autovec-preference=4 -g" \
+      -DCMAKE_C_COMPILER=$(which mpicc) \
+      -DCMAKE_CXX_COMPILER=$(which mpicxx) \
+      -DGMX_OMP=ON \
+      -DGMX_MPI=ON \
+      -DGMX_SIMD=ARM_SVE \
+      -DGMX_BUILD_MDRUN_ONLY=OFF \
+      -DGMX_DOUBLE=OFF \
+      -DCMAKE_INSTALL_PREFIX=${CURDIR} \
+      -DBUILD_SHARED_LIBS=OFF \
+      -DGMX_FFT_LIBRARY=fftw3 \
+      -DFFTWF_LIBRARY=/shared/fftw-3.3.9/lib/libfftw3f.a \
+      -DFFTWF_INCLUDE_DIR=/shared/fftw-3.3.9/include \
+      \
+      -DGMX_BLAS_USER=${ARMPL_LIBRARIES}/libarmpl_lp64.so \
+      -DGMX_LAPACK_USER=${ARMPL_LIBRARIES}/libarmpl_lp64.so \
+      \
+      -DGMX_GPLUSPLUS_PATH=$(which g++) \
+      -DGMXAPI=OFF \
+      -DGMX_GPU=OFF
+```
+
+## Run Gromacs
+slurm script used for running Gromacs job on parallel cluster.
```

Comment:
What is the info we are trying to convey with this slurm script? Just indicate that relevant modules for ArmPL needs to be loaded if that is new info you want to convey.
Also show the perf results obtained with the ArmPL FFT vs FFTW and mention the gain here.